platform-sdk/swirlds-common/src/main/java/com/swirlds/common/wiring/InputWire.java

```diff
@@ -0,0 +1,172 @@
+/*
+ * Copyright (C) 2023 Hedera Hashgraph, LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.swirlds.common.wiring;
+
+import edu.umd.cs.findbugs.annotations.NonNull;
+import edu.umd.cs.findbugs.annotations.Nullable;
+import java.util.Objects;
+import java.util.function.Consumer;
+import java.util.function.Function;
+
+/**
+ * An object that can insert work to be handled by a {@link TaskScheduler}.
+ *
+ * @param <I> the type of data that passes into the wire
+ * @param <O> the type of the data that comes out of the parent {@link TaskScheduler}'s primary output wire
+ */
+public class InputWire<I, O> {
```

Comment:
I understand why you chose "I" and "O", but in general the letters "O", "I", and "L" should never be used for single letter names because different fonts make then difficult to distinguish from "0", "1", and "l".

I suggest using `IN` and `OUT` or different letters.

Reply:
renamed to `IN` and `OUT`